components/eamxx/src/share/tests/coarsening_remapper_tests.cpp

```diff
@@ -90,6 +90,230 @@ void print (const std::string& msg, const ekat::Comm& comm) {
+{
+
+  AbstractGrid::dofs_list_type src_dofs("",nldofs_src);
+  auto src_dofs_h = cmvc(src_dofs);
+  std::iota(src_dofs_h.data(),src_dofs_h.data()+nldofs_src,nldofs_src*comm.rank());
+  Kokkos::deep_copy(src_dofs,src_dofs_h);
+
+  auto src_grid = std::make_shared<PointGrid>("src",nldofs_src,20,comm);
+  src_grid->set_dofs(src_dofs);
+
+  return src_grid;
+}
+
+// Helper function to create fields
+Field
+create_field(const std::string& name, const std::shared_ptr<const AbstractGrid>& grid, const bool twod, const bool vec, const bool mid = false, const int ps = 1)
+{
+  constexpr int vec_dim = 3;
+  constexpr auto CMP = FieldTag::Component;
+  constexpr auto units = ekat::units::Units::nondimensional();
+  auto fl = twod
+          ? (vec ? grid->get_2d_vector_layout (CMP,vec_dim)
+                 : grid->get_2d_scalar_layout ())
+          : (vec ? grid->get_3d_vector_layout (mid,CMP,vec_dim)
+                 : grid->get_3d_scalar_layout (mid));
+  FieldIdentifier fid(name,fl,units,grid->name());
+  Field f(fid);
+  f.get_header().get_alloc_properties().request_allocation(ps);
+  f.allocate_view();
+  return f;
+}
+
+// Helper function to create a remap file
+void create_remap_file(const ekat::Comm& comm, const std::string& filename, std::vector<std::int64_t>& dofs,
+                       const int na, const int nb, const int ns,
+                       const std::vector<Real>& col, const std::vector<Real>& row, const std::vector<Real>& S) 
+{
+
+  scorpio::register_file(filename, scorpio::FileMode::Write);
+
+  scorpio::register_dimension(filename,"n_a", "n_a", na);
+  scorpio::register_dimension(filename,"n_b", "n_b", nb);
+  scorpio::register_dimension(filename,"n_s", "n_s", ns);
+
+  scorpio::register_variable(filename,"col","col","none",{"n_s"},"real","int","int-nnz");
+  scorpio::register_variable(filename,"row","row","none",{"n_s"},"real","int","int-nnz");
+  scorpio::register_variable(filename,"S","S","none",{"n_s"},"real","real","Real-nnz");
+
+  scorpio::set_dof(filename,"col",dofs.size(),dofs.data());
+  scorpio::set_dof(filename,"row",dofs.size(),dofs.data());
+  scorpio::set_dof(filename,"S",  dofs.size(),dofs.data());
+  
+  scorpio::eam_pio_enddef(filename);
+
+  scorpio::grid_write_data_array(filename,"row",row.data(),ns);
+  scorpio::grid_write_data_array(filename,"col",col.data(),ns);
+  scorpio::grid_write_data_array(filename,"S",    S.data(),ns);
+
+  scorpio::eam_pio_closefile(filename);
+}
+
+TEST_CASE("coarsening_remap_nnz>nsrc") {
+  // This is a simple test to just make sure the coarsening remapper works
+  // when the map itself has more remap triplets than the size of the 
+  // source and target grid.  This is typical in monotone remappers from
+  // fine to coarse meshes.
+
+  // -------------------------------------- //
+  //           Init MPI and PIO             //
+  // -------------------------------------- //
+
+  ekat::Comm comm(MPI_COMM_WORLD);
+
+  MPI_Fint fcomm = MPI_Comm_c2f(comm.mpi_comm());
```

Comment:
Not for this PR, perhaps, but we should change our interface fcn, and do the c2f conversion in there. It's kind of silly to have our C interfaces use a Fortran MPI comm (an int)...